asapdiscovery-data/asapdiscovery/data/operators/selectors/mcs_selector.py

```diff
@@ -246,3 +253,148 @@ def _select(
 
     def provenance(self):
         return {"selector": self.dict(), "oechem": oechem.OEChemGetVersion()}
+
+
+class RascalMCESSelector(SelectorBase):
+    """
+    Implementation of the Rascal Maximum Common Edge Subgraphs (MCES) algorithm
+    from the following paper: https://eprints.whiterose.ac.uk/3568/1/willets3.pdf
+
+    See this RDKit blog post for more information:
+    https://greglandrum.github.io/rdkit-blog/posts/2023-11-08-introducingrascalmces.html
+
+    The algorithm is implemented in RDKit as `rdRascalMCES.FindMCES` and is used
+    to find the maximum common substructure between two molecules.
+
+    Works pairwise between a ligand and a complex, and selects the complex with the
+    highest similarity to the ligand.
+    """
+
+    selector_type: ClassVar[str] = "RascalMCESSelector"
+    similarity_threshold: float = Field(
+        0.7,
```

Comment:
this might be a bit tough for larger molecules, perhaps better to do 0.5

Reply:
I'll run a quick check to see what distribution we would expect for our leadlikes, with lots of substituents generally the similarity goes down even though the core is conserved. Can merge for now, can adjust defaults later